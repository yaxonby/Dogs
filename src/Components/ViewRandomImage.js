import React from "react";
import {Component} from "react";
import Fab from '@material-ui/core/Fab';


export default class ViewRandomImage extends Component{
	constructor(props) {
	    super(props);
	    this.state = { AnotherComponent: null,
			               randomFoto: null};
this.loadData=this.loadData.bind(this)
	  }

		loadData() {
				let self=this;
			const { dispatch} = this.props
			fetch("https://dog.ceo/api/breeds/image/random")
				.then(function(response) {
						if (response.status !== 200) {
							console.log('Looks like there was a problem. Status Code: ' +
								response.status);
							return;
						}
						response.json().then(function(data) {
		self.setState({ randomFoto: data.message });
						});
					}
				)
				.catch(function(err) {	console.log('Fetch Error :-S', err)	});
		}

		componentDidMount() {
	    this.loadData()
	  }

	render () {

	//	if (!this.state.randomFoto) { return <div>Loading...</div>
//	  } else {
			console.log(this.state.randomFoto);
	 	 return ( <div className="positionCenter"> <h3> View random image of dog</h3>
			 <div>
<img src={this.state.randomFoto}/>
<Fab color="default" variant='round' onClick={this.loadData}> Next</Fab>
</div>
			 </div>);
//	  };
}}
