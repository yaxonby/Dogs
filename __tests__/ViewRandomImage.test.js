import React from 'react';
import renderer from 'react-test-renderer';
//import Enzyme from 'enzyme';
//import Adapter from 'enzyme-adapter-react-16';
//import { shallow, mount } from 'enzyme';
import ViewRandomImage from '../src/Components/ViewRandomImage';

//Enzyme.configure({ adapter: new Adapter() });

//it("check prop is null by default", () => {
//    const ViewRandomImageComponent = shallow(<ViewRandomImage />);
//    expect(ViewRandomImageComponent.props().dogBreed).toBe(undefined);
//});

it('should render a ViewRandomImage', () => {
    const wrapper = shallow(
        <ViewRandomImage />
    );
    expect(wrapper).toMatchSnapshot();
});
