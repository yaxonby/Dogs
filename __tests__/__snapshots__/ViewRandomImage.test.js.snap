// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a ViewRandomImage 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ViewRandomImage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        " ",
        <p>
           View random image
        </p>,
        <button
          onClick={[Function]}
        >
           Next
        </button>,
        <div>
          <img
            src={null}
          />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      " ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " View random image",
        },
        "ref": null,
        "rendered": " View random image",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " Next",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": " Next",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            src={null}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": null,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          " ",
          <p>
             View random image
          </p>,
          <button
            onClick={[Function]}
          >
             Next
          </button>,
          <div>
            <img
              src={null}
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " View random image",
          },
          "ref": null,
          "rendered": " View random image",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Next",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": " Next",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              src={null}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "src": null,
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
